lib/widgets/content.dart

```diff
@@ -248,25 +248,245 @@ class MessageImage extends StatelessWidget {
+// .sx { color: hsl(120deg 100% 25%); }
+final _kCodeBlockStyleSx = TextStyle(color: const HSLColor.fromAHSL(1, 120, 1, 0.25).toColor());
+// .sr { color: hsl(189deg 54% 49%); }
+final _kCodeBlockStyleSr = TextStyle(color: const HSLColor.fromAHSL(1, 189, 0.54, 0.49).toColor());
+// .s1 { color: hsl(86deg 57% 40%); }
+final _kCodeBlockStyleS1 = TextStyle(color: const HSLColor.fromAHSL(1, 86, 0.57, 0.40).toColor());
+// .ss { color: hsl(241deg 68% 28%); }
+final _kCodeBlockStyleSs = TextStyle(color: const HSLColor.fromAHSL(1, 241, 0.68, 0.28).toColor());
+// .bp { color: hsl(120deg 100% 25%); }
+final _kCodeBlockStyleBp = TextStyle(color: const HSLColor.fromAHSL(1, 120, 1, 0.25).toColor());
+// .vc { color: hsl(241deg 68% 28%); }
+final _kCodeBlockStyleVc = TextStyle(color: const HSLColor.fromAHSL(1, 241, 0.68, 0.28).toColor());
+// .vg { color: hsl(241deg 68% 28%); }
+final _kCodeBlockStyleVg = TextStyle(color: const HSLColor.fromAHSL(1, 241, 0.68, 0.28).toColor());
+// .vi { color: hsl(241deg 68% 28%); }
+final _kCodeBlockStyleVi = TextStyle(color: const HSLColor.fromAHSL(1, 241, 0.68, 0.28).toColor());
+// .il { color: hsl(0deg 0% 40%); }
+final _kCodeBlockStyleIl = TextStyle(color: const HSLColor.fromAHSL(1, 0, 0, 0.40).toColor());
+
 class CodeBlock extends StatelessWidget {
   const CodeBlock({super.key, required this.node});
 
   final CodeBlockNode node;
 
   @override
   Widget build(BuildContext context) {
-    final text = node.text;
-
     return Container(
-      padding: const EdgeInsets.fromLTRB(7, 5, 7, 3),
       decoration: BoxDecoration(
         color: Colors.white,
         border: Border.all(
           width: 1,
-          color: const HSLColor.fromAHSL(0.15, 0, 0, 0).toColor())),
+          color: const HSLColor.fromAHSL(0.15, 0, 0, 0).toColor()),
+        borderRadius: BorderRadius.circular(4)),
       child: SingleChildScrollViewWithScrollbar(
         scrollDirection: Axis.horizontal,
-        child: Text(text, style: _kCodeBlockStyle)));
+        child: Padding(
+          padding: const EdgeInsets.fromLTRB(7, 5, 7, 3),
+          child: _buildText(),
+        )));
+  }
+
+  Widget _buildText() {
+    return switch (node.spans) {
+      []  => Text('', style: _kCodeBlockStyle),
+      [final CodeBlockSpanNode span]
+        when span.tokenType == CodeBlockSpanToken.text
+          => Text(span.text, style: _kCodeBlockStyle),
+      _   => Text.rich(_buildNodes(node.spans)),
+    };
+  }
+
+  InlineSpan _buildNodes(List<CodeBlockSpanNode> nodes) {
+    return TextSpan(
+      style: _kCodeBlockStyle,
+      children: nodes.map(_buildNode).toList(growable: false));
+  }
+
+  InlineSpan _buildNode(CodeBlockSpanNode node) {
+    final style = switch (node.tokenType) {
+      CodeBlockSpanToken.text => null, // styles not applied for Text token
+      CodeBlockSpanToken.highlightedLines => _kCodeBlockStyleHll,
```

Comment:
Hmm, so, here's how this looks for [my highlighted-lines test message](https://chat.zulip.org/#narrow/stream/7-test-here/topic/Code.20blocks/near/1616813):
![image](https://github.com/zulip/zulip-flutter/assets/28173/e065e347-b237-4584-abcb-2cee2ea9adc6)

But here's how it looks on web:
![image](https://github.com/zulip/zulip-flutter/assets/28173/7723e94f-3c7a-4cf0-86d8-13fb297ff7d2)

Note that on web, the text colors and bold are still there — there's just also the background color for the line highlight.

It looks like in order to handle this feature, we'll need some more complication in the parser and the data structure. Here's how the HTML comes out (see `test/model/content_test.dart` for tips on conveniently getting the server's HTML):
```html
<span class="hll"><span class="gh"># he</span>
</span><span class="gu">## llo</span>
```
Note the nesting, with a `span.hll` around another span.

Given that it'd require making our implementation more complicated, and that it's a feature Zulip doesn't support very nicely in the first place (https://github.com/zulip/zulip-flutter/pull/242#issuecomment-1652450667) and it's not clear we actually intend to support, I'd be happy to just not support it, at least for the present. In that case we can have the parser just produce `UnimplementedBlockContentNode` when it encounters one of these.

Let's do include a test case involving this feature, though. That'll help validate that we don't do anything wild when we run into it.

Reply:
Updated the implementation to generically handle one level of nested span nodes in parsing the code block spans. For now, this is sufficient for `span.hll` nodes.

Also added a unit test to handle "highlighted lines" tokens.